shopfloor/actions/data.py

```diff
@@ -86,15 +90,25 @@ def package(self, record, picking=None, with_packaging=False, **kw):
     def packages(self, records, picking=None, **kw):
         return [self.package(rec, picking=picking, **kw) for rec in records]
 
-    @property
-    def _package_parser(self):
-        return [
+    def _package_parser(self, picking=None):
+        parser = [
             "id",
             "name",
-            "pack_weight:weight",
             ("package_storage_type_id:storage_type", ["id", "name"]),
         ]
 
+        context_kwargs = {}
```

Comment:
I'd remove all these changes and add a new computed context dependent field instead (eg: [real|final|shopfloor]_weight) and replace the parser line `pack_weight:weight` w/ `real_weight:weight`. Then when you need to tweak the weight just pass proper ctx keys (eg: `package_real_weight=True`)